### PR TITLE
epicsRingBytes: fix reversed operands in ResetHighWaterMark

### PR DESCRIPTION
**Pull request by jacquelinegarrahan**
The used-bytes count is nextPut - nextGet everywhere else in this file
(epicsRingBytesPut, epicsRingBytesUsedBytes), but
epicsRingBytesResetHighWaterMark computed nextGet - nextPut.  For a ring
holding u bytes it therefore stored size-u; since Put only raises the
mark, the high-water mark stayed permanently inflated and wrong.

Co-Authored-By: Claude Opus 4.8 (1M context) <noreply@anthropic.com>

### CODE REVIEW

File: modules/libcom/src/ring/epicsRingBytes.c [hunk 1]
```diff
@@ -244,7 +244,7 @@ LIBCOM_API void epicsStdCall epicsRingBytesResetHighWaterMark(epicsRingBytesId i
     ringPvt *pring = (ringPvt *)id;
     int used;
     if (pring->lock) epicsSpinLock(pring->lock);
-    used = pring->nextGet - pring->nextPut;
+    used = pring->nextPut - pring->nextGet;
```
Comment (anjohnson) [inline modules/libcom/src/ring/epicsRingBytes.c:247]: My Claude agrees with your Claude that this is a bug, and approves of this fix.

I asked mine why our unit test program didn't discover this. Apparently we're missing any test cases that would show the bug, as currently it only resets the HWM when the ring is empty. I have some extra tests to catch regressions in both ring implementations which I will append to this PR.

**Review by anjohnson — APPROVED** [review]
Thanks for this PR, it also points out where we have some missing tests.

**Review by ralphlange — APPROVED** [review]
LGTM
